### DISCUSSION TITLE
Make php7.2 minimum requirement

### DISCUSSION THREAD

**Discussion by sreichel** [OpenMage/magento-lts#2402] [General]
Long story short ...

- PHP7.2 is end-of-life since Nov 2020
- dev-tools like phpstan require at least PHP7.2
- composer 2.3 requires PHP7.2

Just as we have abandoned support for PHP5, I would like to ....

**Comment by addison74** [comment]
Debian 9 had PHP 7.0 in its repository, but its LTS recently ended on June 30. The PHP packages maintained by Ondrej Sury do not install on Debian 9. Practically, whoever still has this operating system either upgrades to Debian 10 / 11 or continues to use it but without updating (it works but I don't recommend it for a long time). I doubt a developer will use this OS to test OpenMage. Debian 10 comes with PHP 7.3 in its repository and entered LTS recently which will last until 30.06.2024. I use Debian 10 in production and in 1 year I will upgrade to Debian 11/PHP 7.4.

Based on this argument, I think that the minimum requirement for PHP can be changed to 7.3. We should focus our efforts on PHP 8.1 and stay quiet for a while.

**Reply by Sdfendor** [reply]
I would be fine with 7.3 as a step forward. The 8.1 (and soon 8.2) compatibility changes can happen regardless.
